evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -88,3 +93,57 @@ def score(self, X, y, objectives, X_train=None, y_train=None):
             y_pred_proba=None,
             objectives=objectives,
         )
+
+    def get_forecast_periods(self, X):
+        """Generates all possible forecasting periods based on last period of X.
+
+        Args:
+            X (pd.DataFrame, np.ndarray): Data the pipeline was trained on of shape [n_samples_train, n_feautures].
+
+        Raises:
+            ValueError: If pipeline is not trained.
+
+        Returns:
+            pd.Series: Datetime periods out to `forecast_horizon + gap`.
+
+        """
+        if not self._is_fitted:
+            raise ValueError("Pipeline must be fitted before getting forecast.")
+
+        X = infer_feature_types(X)
+
+        # Generate prediction periods
+        first_date = X.iloc[-1][self.time_index]
+        predicted_date_range = pd.Series(
+            pd.date_range(
+                start=first_date,
+                periods=self.forecast_horizon
+                + self.gap
+                + 1,  # Add additional period to account for dropping first date row
+                freq=self.frequency,
+            ),
+        )
+
+        # Generate numerical index
+        first_idx = len(X) - 1 if not isinstance(X.index.dtype, int) else X.index[-1]
```

Comment:
Using this to create a int index that is sequential after X_train

Reply:
Why is this so important to have an integer index?